#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("JointImpedanceController", str_t, 0, "JointImpedanceController", "JointImpedanceController")

gen.add("Joint_Kd", double_t, 0, "Joint_Kd", 0.5,  0, 600)#

gen.add("NocontactImpedance", str_t, 0, "NocontactImpedance", "NocontactImpedance")

gen.add("Cartesian_Md_pos", double_t, 0, "Cartesian_Md_pos", 1, 0, 10)#
gen.add("Cartesian_Dd_pos", double_t, 0, "Cartesian_Dd_pos", 1,  0, 100)#
gen.add("Cartesian_Kd_pos", double_t, 0, "Cartesian_Kd_pos", 150,  0, 500)#

gen.add("Cartesian_Md_ori", double_t, 0, "Cartesian_Md_ori", 1, 0, 10)#
gen.add("Cartesian_Dd_ori", double_t, 0, "Cartesian_Dd_ori", 1,  0, 100)#
gen.add("Cartesian_Kd_ori", double_t, 0, "Cartesian_Kd_ori", 100,  0, 200)#

gen.add("ZK", double_t, 0, "ZK", 30,  0, 200)#

gen.add("VirtualM", double_t, 0, "VirtualM", 0.001, 0, 5)#
gen.add("VirtualD", double_t, 0, "VirtualD", 0.01,  0, 5)#
gen.add("VirtualK", double_t, 0, "VirtualK", 60,  0, 200)#

#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "nocontact_impedance_param"))
